### PR TITLE
Add HTTP transport for synchronous (only) requests

### PR DESCRIPTION
**Pull request by kris-lab**
- [x] support only for synchronous requests

**In first version it will not support for asynchronous request as it requires to have `long-pooling` in place!**

### CODE REVIEW

**Comment by kris-lab** [comment]
@tomaszdurka this is pretty important to pair on it... currently it merges #9 and #12 for simplicity when used with https://github.com/cargomedia/bipbip/pull/132

File: lib/janus_gateway/transport/http.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+        data = JSON.parse(response.body)
+
+        transaction_list = @transaction_queue.clone
+
+        transaction_id = data['transaction']
+        unless transaction_id.nil?
+          promise = transaction_list[transaction_id]
+          unless promise.nil?
+            if %w(success ack).include?(data['janus'])
```
Comment (tomaszdurka) [inline lib/janus_gateway/transport/http.rb]: Didn't you just say that you cannot get ack from sync actions? Or this is actually for sync and async actions ?

Comment (kris-lab) [inline lib/janus_gateway/transport/http.rb]: That's true. For sync actions you cannot get it but with this approach you can also request asynch actions and you will get ACK. However without loon-polling you will not get `event` to resolve transaction completely.

**Comment by kris-lab** [comment]
@tomaszdurka please have a look

**Comment by kris-lab** [comment]
@njam please review

**Comment by njam** [comment]
- _send: On non-200 responses: Also include the response's body in the error, for debugging?
- _send: Remove the assignment of `response_json`, the result of either `if` or `else` will be returned
- send: It's a bit dangerous to run a thread for a synchronous operation without checking the result of the thread. If there's an exception in the thread we won't notice it. It might also hang forever.
 - I understand we want to do this so the operation is fast? What is the specific use case? Could we leave that to the caller (e.g. using https://github.com/igrigorik/em-http-request in `run()` similar to the websocket transport)?
- send: Should `ack` really resolve the promise? I thought this is only to inform the client that the request was received?
- Spec: Use https://github.com/bblimke/webmock ?

**Comment by kris-lab** [comment]
@njam please re-review

**Comment by njam** [comment]
lgtm!

- send: Inline `error_data`?
-  Spec: Use https://github.com/bblimke/webmock ?

**Comment by kris-lab** [comment]
> Spec: Use https://github.com/bblimke/webmock ?

not sure why should we use it?

**Comment by tomaszdurka** [comment]
@njam I understand you would like to test `_send` (as we do not test it) all the way up to ruby http ? I think it's good ida.

**Comment by kris-lab** [comment]
I see your point! Who is up to implement this? I am bit tired in refactoring the same SPEC for 50 times:) once getting @tomaszdurka ideas and once @njam :) 

I am totally OK if you like to finish it up. I would focus on the bipbip?

**Comment by njam** [comment]
Yes, to increase test coverage.
Tomasz volunteered to do it!

**Comment by kris-lab** [comment]
@tomaszdurka thanks. Let me know when is ready. I would like to close the bipbip topic asap.

**Comment by tomaszdurka** [comment]
@kris-lab or @njam please review

**Comment by njam** [comment]
lgtm, just maybe in non-200 also include the http status and the body?

**Comment by tomaszdurka** [comment]
@njam please review

**Comment by njam** [comment]
Hm, the build times out

**Comment by kris-lab** [comment]
@tomaszdurka could you have a look on last commit? There is removed spec. Not sure why it fails on travis with:
```
$ rvm use 2.0.0 --install --binary --fuzzy
Using /home/travis/.rvm/gems/ruby-2.0.0-p598
$ export BUNDLE_GEMFILE=$PWD/Gemfile
$ ruby --version
ruby 2.0.0p598 (2014-11-13 revision 48408) [x86_64-linux]
$ rvm --version
rvm 1.26.10 (latest-minor) by Wayne E. Seguin <wayneeseguin@gmail.com>, Michal Papis <mpapis@gmail.com> [https://rvm.io/]
$ bundle --version
Bundler version 1.7.6
$ gem --version
2.4.5
```

**Comment by tomaszdurka** [comment]
Debugging changes brought this https://github.com/kris-lab/janus-gateway-ruby/compare/kris-lab:20c00f26a9b7dbcc395130a796afa506d89c5e99...5c244fee54031a28a178e1c499391588abe6fb91

**Comment by kris-lab** [comment]
@njam please re-review

**Comment by njam** [comment]
ok, lgtm

**Comment by kris-lab** [comment]
released

**Pull request merged**